crates/subspace-malicious-operator/src/lib.rs

```diff
@@ -19,3 +19,282 @@
+    }
+
+    fn author() -> String {
+        env!("CARGO_PKG_AUTHORS").into()
+    }
+
+    fn support_url() -> String {
+        "https://github.com/subspace/subspace/issues/new".into()
+    }
+
+    fn copyright_start_year() -> i32 {
+        2022
+    }
+
+    fn load_spec(&self, id: &str) -> Result<Box<dyn ChainSpec>, String> {
+        // TODO: Fetch the runtime name of `self.domain_id` properly.
+        let runtime_name = "evm";
+        match runtime_name {
+            "evm" => evm_chain_spec::load_chain_spec(id),
+            unknown_name => Err(format!("Unknown runtime: {unknown_name}")),
+        }
+    }
+}
+
+/// Default sub directory to store network config.
+pub(crate) const DEFAULT_NETWORK_CONFIG_PATH: &str = "network";
+
+/// Create a Configuration object from the current object, port from `sc_cli::create_configuration`
+/// and changed to take `chain_spec` as argument instead of construct one internally.
+pub fn create_malicious_operator_configuration<Cli: SubstrateCli>(
+    domain_cli: &DomainCli,
+    chain_spec: Box<dyn ChainSpec>,
+    tokio_handle: tokio::runtime::Handle,
+) -> sc_cli::Result<Configuration> {
+    let domain_cli_args = &domain_cli.run;
+    let is_dev = domain_cli_args.shared_params().is_dev();
+    let base_path = domain_cli_args
+        .shared_params()
+        .base_path()?
+        .unwrap_or_else(|| BasePath::from_project("", "", &Cli::executable_name()));
+    let config_dir = base_path.config_dir(chain_spec.id());
+    let net_config_dir = config_dir.join(DEFAULT_NETWORK_CONFIG_PATH);
+    let client_id = Cli::client_id();
+    let node_key = domain_cli_args
+        .node_key_params()
+        .map(|x| x.node_key(&net_config_dir))
+        .unwrap_or_else(|| Ok(Default::default()))?;
+    let role = Role::Authority;
+    let max_runtime_instances = 8;
+    let is_validator = role.is_authority();
+    let keystore = domain_cli_args
+        .keystore_params()
+        .map(|x| x.keystore_config(&config_dir))
+        .unwrap_or_else(|| Ok(KeystoreConfig::InMemory))?;
+    let telemetry_endpoints = None;
+    let runtime_cache_size = 2;
+    let network = match domain_cli_args.network_params() {
+        Some(network_params) => network_params.network_config(
+            &chain_spec,
+            is_dev,
+            is_validator,
+            Some(net_config_dir),
+            &client_id,
+            generate_node_name().as_str(),
+            node_key,
+            30334,
+        ),
+        None => NetworkConfiguration::new(
+            generate_node_name().as_str(),
+            &client_id,
+            node_key,
+            Some(net_config_dir),
+        ),
+    };
+
+    Ok(Configuration {
+        impl_name: Cli::impl_name(),
+        impl_version: Cli::impl_version(),
+        tokio_handle,
+        transaction_pool: domain_cli_args.transaction_pool(is_dev)?,
+        network,
+        keystore,
+        // The value just placeholder it will be overwritten later
```

Comment:
Where?

Reply:
https://github.com/subspace/subspace/blob/21fd58be039be4a5c5f33592e63749ec68a99a16/crates/subspace-malicious-operator/src/malicious_domain_instance_starter.rs#L94-L97

Reply:
I see. The comment is a bit misleading though. The only reason path is patched in linked code is because the configuration was created by Substrate's internals and didn't allow for necessary customization. If you create the config directly, I see no reason not to initialize this with proper path from the beginning.